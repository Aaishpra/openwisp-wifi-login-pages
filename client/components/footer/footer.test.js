/* eslint-disable camelcase */
import {shallow} from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Footer from "./footer";

const createTestProps = props => {
  return {
    language: "en",
    footer: {
      links: [
        {
          text: {en: "link one"},
          url: "www.testurl.com",
        },
        {
          text: {en: "link two"},
          url: "www.testurl2.com",
        },
      ],
      secondary_text: {
        en: "this is secondary text",
      },
    },
    ...props,
  };
};

describe("<Footer /> rendering", () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Footer {...props} />);
  });
  it("should render correctly", () => {
    props = createTestProps();
    const component = renderer.create(<Footer {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should render without links", () => {
    const links = {
      footer: {...props.footer, links: []},
    };
    props = createTestProps(links);
    wrapper = shallow(<Footer {...props} />);
    expect(wrapper.find(".owisp-footer-link")).toHaveLength(0);
  });
  it("should render secondary text", () => {
    const {secondary_text} = props.footer;
    expect(wrapper.find(".owisp-footer-row-2-inner").text()).toBe(
      secondary_text.en,
    );
  });
});
