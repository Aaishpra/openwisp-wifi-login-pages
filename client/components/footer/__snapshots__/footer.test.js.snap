// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> rendering should render correctly 1`] = `
<div
  className="owisp-footer-container"
>
  <div
    className="owisp-footer-row-1"
  >
    <div
      className="owisp-footer-row-1-inner"
    >
      <a
        className="owisp-footer-link
                  owisp-footer-link-1"
        href="www.testurl.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        link one
      </a>
      <a
        className="owisp-footer-link
                  owisp-footer-link-2"
        href="www.testurl2.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        link two
      </a>
    </div>
  </div>
  <div
    className="owisp-footer-row-2"
  >
    <div
      className="owisp-footer-row-2-inner"
    >
      this is secondary text
    </div>
  </div>
</div>
`;
